### PR TITLE
feat: Added expire and issued timestamps in attachment

### PR DESCRIPTION
**Pull request by harrrson**
Added methods to extract expiration and issued timestamps from url

```c++
dpp::embed test{};
test.set_title("Test for file")
    .add_field("filename", fileAttachment.filename)
    .add_field("description", fileAttachment.description)
    .add_field("url", fileAttachment.url)
    .add_field("proxy_url", fileAttachment.proxy_url)
    .add_field("Expiration timestamp", std::to_string(fileAttachment.get_expire_time()))
    .add_field("issuing timestamp", std::to_string(fileAttachment.get_issued_time()));
```
obraz

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: src/dpp/message.cpp [hunk 1]
```diff
@@ -882,6 +882,38 @@ bool attachment::is_remix() const {
 	return flags & a_is_remix;
 }
 
+time_t attachment::get_expire_time() const {
+	size_t attr_position = url.find('?');
```
Comment (Jaskowicz1) [inline src/dpp/message.cpp:886]: This can be `const`

File: src/dpp/message.cpp [hunk 2]
```diff
@@ -882,6 +882,38 @@ bool attachment::is_remix() const {
...
+	/*If no attributes were sent in url, we do not need to parse more*/
+	if(url.npos == attr_position){
+		return 0;
+	}
+	std::string attributes = url.substr(attr_position+1);
+	std::vector<std::string> attr_list = utility::tokenize(attributes,"&");
```
Comment (Jaskowicz1) [inline src/dpp/message.cpp:892]: imo, you should put `url.substr(attr_position+1)` inside `utility::tokenize` as you don't use `attributes` anywhere else but this. However, I'd like other thoughts on this.

File: src/dpp/message.cpp [hunk 3]
```diff
@@ -882,6 +882,38 @@ bool attachment::is_remix() const {
 	return flags & a_is_remix;
 }
 
+time_t attachment::get_expire_time() const {
+	size_t attr_position = url.find('?');
+	/*If no attributes were sent in url, we do not need to parse more*/
```
Comment (Jaskowicz1) [inline src/dpp/message.cpp:887]: Space the comments correctly!
`/* If ......... */`

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: src/dpp/message.cpp (hunk 1 above)
Comment (Mishura4) [reply src/dpp/message.cpp:886]: which we don't enforce

Comment (harrrson) [reply src/dpp/message.cpp:886]: Yeah, generally i do not use const in local variables (no particular reason, just forgetting to use it). I can add it if needed

File: src/dpp/message.cpp (hunk 2 above)
Comment (braindigitalis) [reply src/dpp/message.cpp:892]: please space stuff out, (attr_position + 1) not (attr_position+1)

File: src/dpp/message.cpp (hunk 1 above)
Comment (Jaskowicz1) [reply src/dpp/message.cpp:886]: > Yeah, generally i do not use const in local variables (no particular reason, just forgetting to use it). I can add it if needed

No need :)

File: src/dpp/message.cpp (hunk 2 above)
Comment (harrrson) [reply src/dpp/message.cpp:892]: I generally like to add intermediate variables to make the code easier to follow, iirc compiler optimizes them away if able to

File: src/dpp/message.cpp (hunk 3 above)
Comment (harrrson) [reply src/dpp/message.cpp]: Done

**Review by braindigitalis — APPROVED** [review]
approved and good to merge, would be nice if there were unit tests for this though

**Comment by Jaskowicz1** [comment]
> approved and good to merge, would be nice if there were unit tests for this though

A todo list for stuff would be lovely

**Review by Jaskowicz1 — APPROVED** [review]

**Pull request merged**